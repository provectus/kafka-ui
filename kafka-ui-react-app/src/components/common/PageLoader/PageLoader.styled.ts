import styled from 'styled-components';

export const Wrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  padding-top: 15%;
  height: 100%;
  width: 100%;
`;
