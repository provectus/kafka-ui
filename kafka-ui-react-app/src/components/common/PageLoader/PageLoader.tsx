import React from 'react';
import cx from 'classnames';

interface Props {
  fullHeight: boolean;
}

const PageLoader: React.FC<Partial<Props>> = ({ fullHeight }) => (
  <section
    className={cx(
      'hero',
      fullHeight ? 'is-fullheight-with-navbar' : 'is-halfheight'
    )}
  >
    <div
      className="hero-body has-text-centered"
      style={{ justifyContent: 'center' }}
    >
      <div style={{ width: 300 }}>
        <div className="subtitle">Loading...</div>
        <progress
          className="progress is-small is-primary is-inline-block"
          max="100"
        />
      </div>
    </div>
  </section>
);

export default PageLoader;
