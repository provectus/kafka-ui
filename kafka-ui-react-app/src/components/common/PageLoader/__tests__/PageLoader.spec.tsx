import { mount } from 'enzyme';
import React from 'react';
import PageLoader from '../PageLoader';

describe('PageLoader', () => {
  it('matches the snapshot', () => {
    expect(mount(<PageLoader />)).toMatchSnapshot();
  });

  it('renders half-height page loader by default', () => {
    const wrapper = mount(<PageLoader />);
    expect(wrapper.exists('.hero.is-halfheight')).toBeTruthy();
    expect(wrapper.exists('.hero.is-fullheight-with-navbar')).toBeFalsy();
  });

  it('renders fullheight page loader', () => {
    const wrapper = mount(<PageLoader fullHeight />);
    expect(wrapper.exists('.hero.is-halfheight')).toBeFalsy();
    expect(wrapper.exists('.hero.is-fullheight-with-navbar')).toBeTruthy();
  });
});
