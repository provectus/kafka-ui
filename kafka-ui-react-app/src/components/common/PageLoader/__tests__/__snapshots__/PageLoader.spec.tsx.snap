// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLoader matches the snapshot 1`] = `
<PageLoader>
  <section
    className="hero is-halfheight"
  >
    <div
      className="hero-body has-text-centered"
      style={
        Object {
          "justifyContent": "center",
        }
      }
    >
      <div
        style={
          Object {
            "width": 300,
          }
        }
      >
        <div
          className="subtitle"
        >
          Loading...
        </div>
        <progress
          className="progress is-small is-primary is-inline-block"
          max="100"
        />
      </div>
    </div>
  </section>
</PageLoader>
`;
